### PR TITLE
Add check for private image

### PR DESCRIPTION
**Pull request by StephenAraka**
### What does this PR do?
Adds checkbox for private image... and if checked, renders required input for dockerhub private images...

Peek 2020-05-13 16-26

### What is happening...
Check if the checkbox is checked and show the tab

### This PR also...
- Adds a type prop to the `BlackInputText` component, this is used to check if the input is of type password.
- Sets modal `maximum height` to 80% and makes the overflow content `vertically scrollable`...

### CODE REVIEW

File: src/components/AppsPage/index.jsx [hunk 1]
```diff
@@ -260,8 +323,68 @@ class AppsPage extends React.Component {
...
+                            name="email"
+                            value={email}
+                            onChange={(e) => {
+                              this.handleDockerCredentialsChange(e);
+                            }}
+                          />
+
+                          <BlackInputText
+                            required
+                            placeholder="Password"
```
Comment (neelxie) [inline src/components/AppsPage/index.jsx:362]: @StephenAraka shouldn't this placeholder be **Docker Password**

**Comment by StephenAraka** [comment]
> @StephenAraka I noticed after the private image checkbox is clicked, you could as well have some validation that ensures that as long as the checkbox is ticked the preceding input field could have the relevant inputs on submission, unlike now where it is ticked and a successful submission is possible yet none of the details for the private image were entered.

Thanks for the review Derek. If u read the PR description. I mentioned that issue...
According to our workflow, I am trying to avoid doing too much in one Pull Request.

File: src/components/AppsPage/index.jsx (hunk 1 above)
Comment (StephenAraka) [reply src/components/AppsPage/index.jsx:362]: Alright... let me change that.

**Comment by neelxie** [comment]
> > @StephenAraka I noticed after the private image checkbox is clicked, you could as well have some validation that ensures that as long as the checkbox is ticked the preceding input field could have the relevant inputs on submission, unlike now where it is ticked and a successful submission is possible yet none of the details for the private image were entered.
> 
> Thanks for the review @neelxie . If u read the PR description. I mentioned that issue...
> According to our workflow, I am trying to avoid doing too much in one Pull Request.
> Should I go ahead and add validation?

If you already have a PR already then its okay kindly continue

**Comment by StephenAraka** [comment]
> If you already have a PR already then its okay kindly continue

I don't have a PR already. The plan was to do validation after this one is merged. 
Or should I do validation on this one?

**Comment by inno-asiimwe** [comment]
> > If you already have a PR already then its okay kindly continue
> 
> I don't have a PR already. The plan was to do validation after this one is merged.
> Or should I do validation on this one?

If it is part of this feature, then add it to the same PR.

**Comment by StephenAraka** [comment]
> > > If you already have a PR already then its okay kindly continue
> > 
> > 
> > I don't have a PR already. The plan was to do validation after this one is merged.
> > Or should I do validation on this one?
> 
> If it is part of this feature, then add it to the same PR.

Cool... Lemme do that

**Pull request merged**